cpp/src/arrow/compute/kernels/scalar_validity_test.cc

```diff
@@ -77,17 +77,100 @@ TEST_F(TestBooleanValidityKernels, ArrayIsNull) {
                    "[true, false, false, true]");
 }
 
+TEST_F(TestFloatValidityKernels, FloatArrayIsNull) {
+  // All NaN
```

Comment:
I suggest to combine all the `TestFloatValidityKernels` tests into a single block and similarly for the `TestDoubleValidityKernels`. You can use comments to specify the type of test performed and change the options directly (e.g., `options.nan_is_null = false`).

Reply:
Addressed on 915c868 .